numpy/lib/npyio.py

```diff
@@ -253,6 +255,138 @@ def iterkeys(self):
+    file resulting in one concatenated array saved. Saved arrays must
+    have exactly the same dtype and have compatible shapes. Shapes can
+    differ in the 1st dimension only which will be added up in the file.
+    The file is written in C order.
+
+    Parameters
+    ----------
+    fp : file-like object or string
+        Name of the file to open or a file-like object supporting
+        ``tell()``, ``seek()`` and ``write()``. In the latter case,
+        data will be written from current position.
+    flush : bool, optional
+        Calls ``flush()`` at the end of each ``save()``. Default:
+        False
+    hdrupdate : bool, optional
+        If True, updates header after each call to ``save()`` by seeking back
+        to the beginning (starting position) of the file, rewrites
+        header then seeks to the current position again.  Otherwise header
+        will be updated once at ``close()``.  Default: True.
+    version : tuple or None, optional
+       The .npy file version to use or to detect automatically at
+       first ``save()`` call if None.
+
+    See Also
+    --------
+    load
+    """
+    #TODO: implement header reading to open for append and for partial
+    #reading.
+
+    EXTRAPAD=32 # We allow 32 bytes to grow in the shape description
+
+    def __init__(self, fp, flush=False, hdrupdate=True,version=None):
+        if isfileobj(fp):
+            self.fp = fp
+            self.hfilepos = fp.tell()
+        else:
+            self.fp = open(fp, 'wb')
+            self.hfilepos = 0
+        self.flush = flush
+        self.hdisklen = 0
+        self.hdrupdate = hdrupdate
+        self.version = version
+
+    def save(self, arr,headercheck=True):
+        """
+        Append an array to the ``.npy`` format binary file. The first call of
+        ``save()`` determines the dtype description to be written into
+        the file.  `arr` must have the same dtype and shape with the
+        exception of the 1st dimension in all subsequent calls. The file is
+        written in C order.
+
+        Parameters
+        ----------
+        arr : array-like
+            Array to be saved.
+        headercheck : bool, optional
+            Whether to check dtype and shape compatibility with the
+            first call. Setting to False allows incompatible array
+            data to be appended that will corrupt the file. Default: True.
+        """
+        arr=np.asanyarray(arr)
+        # In case of F contiguous arrays, tofile() would produce a F
+        # order file.  At the moment, we support the shape update for
+        # C order only. If arr is neither C nor F contiguous, then
+        # tofile() will generate C order (same approach in
+        # format.write_array)
+        if arr.dtype.hasobject:
+            raise TypeError("Object arrays cannot be appended in .npy format")
+        if arr.flags.f_contiguous and not arr.flags.c_contiguous:
+            arr = np.asanyarray(arr,order='C')
```

Comment:
I don't understand the motivation for the flags check here -- don't we just want to require `C` order unconditionally?